services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,215 @@
+	// Construct the GoogleCredentials object which obtains the default configuration from your
+	// working environment.
+	ts, err := idtoken.NewTokenSource(ctx, url, option.WithCredentialsJSON([]byte(destConfig.Credentials)))
+	if err != nil {
+		fmt.Print("failed to create NewTokenSource: %w", err)
+	}
+
+	// Get the ID token, to make an authenticated call to the target audience.
+	token, err := ts.Token()
+	if err != nil {
+		fmt.Print("failed to receive token: %w", err)
+	}
+
+	// Create a POST request
+	req, err := http.NewRequest("POST", destConfig.GoogleCloudFunctionUrl, bytes.NewBuffer(jsonData))
+	if err != nil {
+		fmt.Println("Error creating request:", err)
+		return
+	}
+
+	// Set the appropriate headers
+	req.Header.Set("Content-Type", "application/json")
+	if destConfig.RequireAuthentication {
+		req.Header.Add("Authorization", "Bearer "+token.AccessToken)
+	}
+
+	// Make the request using the client
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		fmt.Println("Error making request:", err)
+		return
+	}
+
+	if resp.Status == "OK" {
+		respStatus = "Success"
+		responseMessage = "[GoogleSheets] :: Message Payload inserted with messageId :: " + parsedJSON.Get("id").String()
+	}
+	fmt.Print(resp.Status)
+
+	return 200, respStatus, responseMessage
+}
+
+// Initialize the Cloud Functions API client using service account credentials.
+func generateService(destination *backendconfig.DestinationT, opts ...option.ClientOption) (*cloudfunctions.Service, error) {
+	ctx := context.Background()
+
+	cloudFunctionService, err := cloudfunctions.NewService(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("[GoogleCloudFunction] error  :: Unable to create cloudfunction service :: %w", err)
+	}
+	return cloudFunctionService, err
+}
+
+func getFunctionName(url string) (functionName string) {
+
+	// Define a regular expression pattern to match URLs between "https://" and dot (.)
+	pattern := `https://(.*?)\.`
+
+	// Compile the regular expression pattern
+	regExp := regexp.MustCompile(pattern)
+
+	// Find all matches in the input string
+	matches := regExp.FindAllStringSubmatch(url, -1)
+
+	// Extract the captured groups
+	var resultArray []string
+	for _, match := range matches {
+		if len(match) > 1 {
+			resultArray = append(resultArray, match[1])
+		}
+	}
+
+	// Combine the first two elements
+	splitValues := strings.Split(resultArray[0], "-")
+	// Join the first two values with "-" and assign to REGION
+	REGION := strings.Join(splitValues[:2], "-")
+
+	// Join the rest of the values with "-" and assign to PROJECT_ID
+	PROJECT_ID := strings.Join(splitValues[2:], "-")
+
+	var FUNCTION_NAME string
+
+	index := strings.Index(url, "cloudfunctions.net/")
+
+	if index != -1 {
+		// Extract the substring after "cloudfunctions.net/"
+		FUNCTION_NAME = url[index+len("cloudfunctions.net/"):]
+	} else {
+		fmt.Println("No match found")
+	}
+
+	functionName = "projects/" + PROJECT_ID + "/locations/" + REGION + "/functions/" + FUNCTION_NAME
```

Comment:
Can't we get each of these separately as config/event(dynamic config) fields ?

Reply:
(Or)
Is there any other easy way to get each of these values from `googleCloudFunctionUrl` field ?

Reply:
Couldn't find a simpler way to get these from the URL. If we ask these all from config, it will be kind of repetitive.